python/cudf/tests/test_hdf.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+import cudf
+from cudf.tests.utils import assert_eq
+
+import pytest
+import os
+import pandas as pd
+import numpy as np
+from string import ascii_letters
+
+
+try:
+    import tables # noqa F401
+    _have_pytables = True
+except ImportError:
+    _have_pytables = False
+if not _have_pytables:
+    pytest.skip('PyTables is not installed and is required for HDF '
+                'reading/writing', allow_module_level=True)
+
+
+@pytest.fixture(params=[0, 1, 10, 100])
+def pdf(request):
+    types = ['bool', 'int8', 'int16', 'int32', 'int64', 'float32', 'float64',
+             'datetime64[ms]']
+    renamer = {'C_l0_g' + str(idx): 'col_' + val for (idx, val) in
+               enumerate(types)}
+    typer = {'col_' + val: val for val in types}
+    ncols = len(types)
+    nrows = request.param
+
+    # Create a pandas dataframe with random data of mixed types
+    test_pdf = pd.util.testing.makeCustomDataframe(
+        nrows=nrows,
+        ncols=ncols,
+        data_gen_f=lambda r, c: r,
+        r_idx_type='i'
+    )
+    # Delete the name of the column index, and rename the row index
+    del(test_pdf.columns.name)
+    test_pdf.index.name = 'test_index'
+
+    # Cast all the column dtypes to objects, rename them, and then cast to
+    # appropriate types
+    test_pdf = test_pdf.astype('object')\
+                       .rename(renamer, axis=1)\
+                       .astype(typer)\
+                       .rename({'col_datetime64[ms]': 'col_datetime64'},
+                               axis=1)
+
+    # Create non-numeric categorical data otherwise may be typecasted
+    data = [ascii_letters[np.random.randint(0, 52)] for i in
+            range(nrows)]
+    test_pdf['col_category'] = pd.Series(data, dtype='category')
+
+    return (test_pdf, nrows)
```

Comment:
The duplication of this and the above `pdf` fixture gives me minor heebie jeebies. It might not be worth trying to factor them together. Some kind of long term fantasy to combine these into utils? Just a comment, no change required.

Reply:
I agree we should build better testing utils and standardize testing around them, but I viewed that as a longer term effort that I didn't want to try to tackle in this PR. I'll create a github issue for us to follow up on in the future.

Reply:
Issue created here: https://github.com/rapidsai/cudf/issues/951